File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, string argv[]){
    if(argc !=2) {
    printf("Usage: ./caesar k\n");
    return 1;
        
    }
   
    int k = atoi(argv[1]);
    printf("plaintext: ");
    string p = get_string();
    for(int i =0;i< strlen(p);i++){
        if(islower(p[i])){
           int d = p[i]-'a';
            d = (d+k)%26;
            p[i] = 'a' + d;
        }
        else if (isupper(p[i])){
            int d = p[i]-'A';
            d = (d+k)%26;
            p[i] = 'A' + d;
        }
        
    }
    printf("ciphertext: %s\n", p);
  
}

File: caesar.py
import sys
import cs50

argc  = len(sys.argv)
argl  = list(sys.argv)
if(argc !=2):
    print("Usage: ./caesar k")
    exit(1)

k= int(argl[1])
#print(k)
print("plaintext: ",end ='')
p = cs50.get_string()
q=str()
for i in range(0,len(p)):
 #   print(type(p[i])," ",end='')
    if(p[i].islower()):
        d = ord(p[i]) - ord('a')
        d = (d+k)%26
        q =q+ chr(ord('a') + d)
        
    elif (p[i].isupper()):
        d = ord(p[i])-ord('A')
        d = (d+k)%26
        q =q+ (chr(ord('A') + d))
    else:
        q = q+p[i]
  #  print(q[i]," ",end='')    
    
print("ciphertext: ", q)


    